### ISSUE TITLE
Watermark placement

### ISSUE THREAD

**Issue by bossanova808** [aelvan/Imager-Craft#18]
This is more of a feature request than an issue as such.

I would love it if watermarks were available to use in the effects {} area.  

Practical use case - master image is a huge 4K image, around 4000 by 2000.  I want to resize for use in a masthead, and then watermark in the center of the resized image.  At the moment I have to chain two transforms to achieve this becuase of course I don't know the dimensions of the resulting resized image until after that has occured.

Ideally, I'd be able to do something like

`effects : { watermark: { postion {50%, 50%}}}`

No problems of course if that's too hard, but I think watermarking transformed images in general would be very useful for people.

**Comment by aelvan** [comment]
Rerlative positioning of the watermark is on my todo-list. :)

In regards to the watermark being an effect, I'm not 100% sure I understand why that would make a difference. Is it so that the watermark is added before the transform actually takes place?

**Comment by bossanova808** [comment]
Ah great.  

Yes, the issues is I can't put a watermark in the centre of a resized image without either it being applied to the image before resizing (when I know the dimensions, or could do it with relative positioning ideally) - or knowing the dimensions of the resized image.  

At the moment it is applied after resizing and only with absolute pixels, so I have to first resize in one transform, get dimensions from that, and then watermark once I know the dimensions.

(Logically I expected that yep, the watermark would be applied before the resize and thus appear in the middle of the resized, which you could currently do if it was available as an effect)

Basically, many ways to skin this cat but the ideal results is a resize and watermark in the centre of the resized image with one transform, whichever way you think works best really!

**Comment by aelvan** [comment]
Yeah, I think the solution is to allow relative positioning on the watermark, and to be able to set the watermark before the resize is done, as with `preEffects` (which is added before resize, whereas `effects` is added after).

**Comment by bossanova808** [comment]
Yes, thanks - I thinks that's a good plan.
